todo/src/App.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from "react";
+import logo from "./logo.svg";
+import "./App.css";
+import Todos from "./components/Todos";
+import TodoForm from "./components/TodoForm";
+import { connect } from "react-redux";
+import { addTodo } from './actions';
+import { deleteTodo } from './actions';
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todoText: ""
+    };
+  }
+
+  handleInputChange = event => {
+    this.setState({ [event.target.name]: event.target.value });
+  };
+
+  addTodo = () => {
+    const { todoText } = this.state;
+    const newTodo = {
+      id: this.props.todos.length + 1,
```

Comment:
This is neat, but I can see it causing problems. If you delete a todo, and then add one, it's possibly (and likely) that two todos are going to have the same id. Let's say you have three todos, so with this method, their ids are 1, 2, 3, and then let's say you delete the first one, so you have 2, 3. Now you add a todo, and with this logic, you'll end up with 2, 3, 3. See the issue? So use `uuid` or `Date.now`.